### DISCUSSION TITLE
Eglot `eglot-ensure` doesn't work on some mode hooks (but not others)

### DISCUSSION THREAD

**Discussion by alexispurslane** [joaotavora/eglot#1399] [General]
So I've got a couple file types set up to eglot-ensure:

```elisp
(add-hook 'rust-ts-mode-hook 'eglot-ensure)
(add-hook 'typescript-ts-mode-hook 'eglot-ensure)
(add-hook 'js-ts-mode-hook 'eglot-ensure)
(add-hook 'python-ts-mode-hook 'eglot-ensure)
```

This works perfectly fine with Rust and Python, but with JavaScript and TypeScript files, nothing happens. No errors, but also no eglot. Nothing whatsoever at all. But if I then manually run `M-x eglot`, sure enough, it works just f-n fine.

**Comment by manuel-uberti** [comment]
It's hard to tell what is going on given only `add-hook` statements.

Since `M-x eglot` works as expected, this does not seem to be a problem related to Eglot itself, but reading the [Troubleshooting](https://joaotavora.github.io/eglot/#Troubleshooting-Eglot) section of the manual would not hurt.

Would you be able to provide a recipe from `emacs -Q` to reproduce the problem?

**Reply by joaotavora** [reply]
Thanks Manuel :)

**Reply by alexispurslane** [reply]
> Would you be able to provide a recipe from emacs -Q to reproduce the problem?

Yes actually, since I was able to repro this problem with a completely clean emacs config — give me a bit and I'll provide it. Apologies for the lack of clarity

**Reply by alexispurslane** [reply]
Also, eglot ensure actually doesn't work with rust or python either on that fresh config, so at least it is consistent.

**Reply by manuel-uberti** [reply]
@alexispurslane any news on the recipe from `emacs -Q`?

**Reply by alexispurslane** [reply]
Oh shit I totally forgot, I got so used to manually triggering eglot!

Okay, upon trying it out, now it's working in both `emacs -Q` and my config. Not sure what the difference is. Apologies!

**Reply by joaotavora** [reply]
> I got so used to manually triggering eglot!

Yeah :-D  Personally I never use eglot-ensure myself, once you realize that `M-x eglot` survives all the way to the lifetime of a project, you probably don't mind typing it, like, once a day.  `eglot-ensure` is supported, but is much more brittle and error-prone.

> Okay, upon trying it out, now it's working in both emacs -Q and my config. Not sure what the difference is. Apologies

Good to know
